.github/workflows/render-site.yaml

```diff
@@ -0,0 +1,49 @@
+name: Check rendered site for broken links
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
```

Comment:
I'd add a statement here that exludes draft PRs (runs the workflow only for PRs that are ready for review). You can look at https://github.com/validmind/documentation/blob/main/.github/workflows/release_notes_check.yml#L5 for an example of how that's done, the `ready_for_review` bit specifically.